libbeat/processors/actions/rename_fields.go

```diff
@@ -0,0 +1,69 @@
+package actions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type renameFields struct {
+	Fields []FieldTuple
+}
+
+type FieldTuple struct {
+	From string `config:"from"`
+	To   string `config:"to"`
+}
+
+func init() {
+	processors.RegisterPlugin("rename_fields",
+		configChecked(newRenameFields,
+			requireFields("fields")))
+}
+
+func newRenameFields(c *common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []FieldTuple `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the rename_fields configuration: %s", err)
+	}
+
+	// TODO: Check if some fields are not allowed to be renamed
+
+	f := &renameFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f *renameFields) Run(event *beat.Event) (*beat.Event, error) {
+
+	for _, field := range f.Fields {
+		value, err := event.Fields.GetValue(field.From)
+		if err != nil {
+			logp.Err("Could not fetch from key: %s", field.From)
```

Comment:
Every key that does not exists will be an error. That seems to verbose to me. Maybe you want to handle the error types separately -- log a debug message if the key does not exists and log an error for anything else.

Reply:
I think adding the `ignore_missing` config option should solve this? There is still the issue of what if there is an "other" error. Need to check if any other errors can happen.